NGINX Plus: dynamic upstream reloads support (#1469)
nginxinc/nginx-gateway-fabric: site/content/overview/gateway-architecture.md

1. (HTTP, HTTPS) A _client_ sends traffic to and receives traffic from any of the _NGINX workers_ on ports 80 and 443.
1. (HTTP, HTTPS) An _NGINX worker_ sends traffic to and receives traffic from the _backends_.

## Updating upstream servers

> 
> The normal process to update any changes to NGINX, as described previously, is to write the configuration files and reload NGINX. When using NGINX Plus, however, we can take advantage of the [NGINX Plus API](http://nginx.org/en/docs/http/ngx_http_api_module.html) to limit the amount of reloads that we have to trigger when making changes to NGINX. Specifically, when the endpoints of an application in Kubernetes change (scaling up or down, for example), the NGINX Plus API is used to update the upstream servers in NGINX with the new endpoints, without the need for a reload. This reduces the potential for a disruption that could occur when reloading.
> 

Reviewer: ```suggestion

The normal process to update any changes to NGINX is to write the configuration files and reload NGINX. However, when using NGINX Plus, we can take advantage of the [NGINX Plus API](http://nginx.org/en/docs/http/ngx_http_api_module.html) to limit the amount of reloads triggered when making changes to NGINX. Specifically, when the endpoints of an application in Kubernetes change (Such as scaling up or down), the NGINX Plus API is used to update the upstream servers in NGINX with the new endpoints without a reload. This reduces the potential for a disruption that could occur when reloading.

```
Small phrasing changes.
